src/edit/classificationFields.test.js

```diff
@@ -0,0 +1,62 @@
+import PropTypes from 'prop-types';
+import { fireEvent, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import '../../test/jest/__mock__';
+
+import stripesFinalForm from '@folio/stripes/final-form';
+import renderWithRouter from '../../test/jest/helpers/renderWithRouter';
+import renderWithIntl from '../../test/jest/helpers/renderWithIntl';
+
+import ClassificationFields from './classificationFields';
+import translationsProperties from '../../test/jest/helpers/translationsProperties';
+
+jest.unmock('@folio/stripes/components');
+
+const onSubmit = jest.fn();
+
+const props = {
+  classificationTypes: [{ name: 'classificationName', id: '369369' }]
+};
+
+const Form = ({ handleSubmit }) => (
+  <form onSubmit={handleSubmit}>
+    <ClassificationFields {...props} />
+  </form>
+);
+
+Form.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+};
+
+const WrappedForm = stripesFinalForm({
+  canAdd: true,
+  canEdit: true,
+  canDelete: true,
+})(Form);
+
+const renderClassificationFields = () => renderWithIntl(
+  renderWithRouter(<WrappedForm onSubmit={onSubmit} />),
+  translationsProperties,
+);
+
+afterEach(() => jest.clearAllMocks());
+
+describe('ClassificationFields', () => {
+  it('renders RepeatableField', () => {
+    renderClassificationFields();
+    expect(screen.getByText('Classification')).toBeInTheDocument();
+  });
+
+  it('click on Add classification button and check dropdown length, enter value in text box', () => {
+    renderClassificationFields();
+    const classificationButton = screen.getByText('Add classification');
+    userEvent.click(classificationButton);
+    const ClassificationDropdown = screen.getAllByText('Select classification type');
+    expect(ClassificationDropdown).toHaveLength(1);
+    const inputText = screen.getByRole('textbox');
+    expect(inputText).toHaveValue('');
+    fireEvent.change(inputText, { target: { value: 'Enter text' } });
+    expect(inputText).toHaveValue('Enter text');
```

Comment:
`userEvent` is preferable to `fireEvent` since it better mimics how a user actually interacts with the browser. Please don't mix them, or if you do mix them, explain why this was necessary in a comment.

Reply:
Hi Zak, userEvent is used and changes have done.